--- /usr/share/jolla-calendar/pages/DatePickerPanel.qml
+++ /usr/share/jolla-calendar/pages/DatePickerPanel.qml
@@ -3,7 +3,7 @@ import Sailfish.Silica 1.0
 import org.nemomobile.calendar 1.0
 import org.nemomobile.time 1.0
 import "Util.js" as Util
-
+import "LunarCalendar.js" as Lunar
 Column {
     id: datePickerContainer
 
@@ -103,7 +103,12 @@ Column {
 
                     Text {
                         id: label
-                        anchors.centerIn: parent
+                        anchors{
+                            top:parent.top
+                            topMargin:Theme.paddingSmall
+                            horizontalCenter: parent.horizontalCenter
+
+                        }
                         text: model.day
                         font.pixelSize: Theme.fontSizeMedium
                         font.bold: model.day === wallClock.time.getDate()
@@ -122,11 +127,72 @@ Column {
                             return Theme.secondaryColor
                         }
                     }
+                    Label{
+                        id:lunarday
+                        anchors{
+                            top:label.bottom
+			                      topMargin:Theme.paddingSmall*0.3
+                            horizontalCenter: parent.horizontalCenter
+                        }
+                        text: {
+                              var lunar = Lunar.LunarCalendar.solarToLunar(model.year, model.month, model.day)
+
+                              var lunarDay = lunar.lunarDay
+                              if(lunarDay == 1||lunarDay == "1"){
+                                  return lunar.lunarMonthName
+                              }
+                              return lunar.lunarDayName
+                            }
+                        font.pixelSize: Theme.fontSizeSmall*0.8
+                        font.bold: model.day === wallClock.time.getDate()
+                                    && model.month === wallClock.time.getMonth()+1
+                                    && model.year === wallClock.time.getFullYear()
+                        color: {
+                            if (model.day === datePicker.day &&
+                                model.month === datePicker.month &&
+                                model.year === datePicker.year) {
+                                return Theme.highlightColor
+                            } else if (lunarday.font.bold) {
+                                return Theme.highlightColor
+                            } else if (model.month === model.primaryMonth) {
+                                return Theme.primaryColor
+                            }
+                            return Theme.secondaryColor
+                        }
+                    }
+                    Rectangle {
+                        id:workday
+                        anchors{
+                          top:label.baseline
+                          topMargin: 5
+                          right:parent.right
+                          rightMargin: events.count > 0?parent.width/2/2:parent.width/2 - parent.width / 5 /2
+                        }
+                        width: parent.width / 5
+                        height: 4
+                        radius: 2
+                        visible: {
+                          var lunar = Lunar.LunarCalendar.solarToLunar(model.year, model.month, model.day)
+                          return lunar.worktime != 0||lunar.worktime != "0"
+                        }
+                        color: {
+                          var lunar = Lunar.LunarCalendar.solarToLunar(model.year, model.month, model.day)
+                          var worktime = lunar.worktime;
+                          if(worktime == 2){
+                            return "#00FF7F"
+                          }else if(worktime == 1){
+                            return "#FA8072"
+                          }else{
+                            return "#00000000"
+                          }
 
+                        }
+                    }
                     Rectangle {
                         anchors.top: label.baseline
                         anchors.topMargin: 5
-                        anchors.horizontalCenter: parent.horizontalCenter
+                        anchors.left: parent.left
+                        anchors.leftMargin:workday.visible?parent.width/2/2:parent.width/2 - parent.width / 5 /2
                         width: parent.width / 5
                         radius: 2
                         height: 4
--- /usr/share/jolla-calendar/pages/MonthPage.qml
+++ /usr/share/jolla-calendar/pages/MonthPage.qml
@@ -3,7 +3,7 @@ import Sailfish.Silica 1.0
 import org.nemomobile.calendar 1.0
 import org.nemomobile.time 1.0
 import "Util.js" as Util
-
+import "LunarCalendar.js" as Lunar
 Page {
     id: root
 
@@ -23,6 +23,41 @@ Page {
         pageStack.push("EventEditPage.qml", { defaultDate: d })
     }
 
+    function getLunarInfo(){
+      var now = new Date();
+      var d = datePicker.date
+      if(!d.getFullYear()){//防止坑爹的jolla打开日历当天Date竟然是NAN的问题
+        d = now;
+      }
+      return Lunar.LunarCalendar.solarToLunar(d.getFullYear(),d.getMonth()+1,d.getDate())
+    }
+    function showLunarInfo(){
+      var info = getLunarInfo();
+      var zodiac = info.zodiac;
+      var GanZhiYear = info.GanZhiYear;
+      var solarFestival = info.solarFestival;
+      var lunarFestival = info.lunarFestival;
+      var term = info.term;
+      var lunarMonthName = info.lunarMonthName;
+      var lunarDayName = info.lunarDayName;
+      var showtext=GanZhiYear+"["+zodiac+"]年 "+lunarMonthName+lunarDayName;
+      return showtext;
+    }
+    function showFestival(){
+      var info = getLunarInfo();
+      var solarFestival = info.solarFestival;
+      var lunarFestival = info.lunarFestival;
+      var term = info.term;
+      var showtext="";
+      if(solarFestival)showtext+=solarFestival+",";
+      if(lunarFestival)showtext+=lunarFestival+",";
+      if(term){
+        showtext+=term;
+      }else{
+        showtext=showtext.substring(0,showtext.length-1)
+      }
+      return showtext;
+    }
     states: State {
         name: "hidePageStackIndicator"
         when: root.status != PageStatus.Inactive
@@ -39,7 +74,7 @@ Page {
 
     SilicaListView {
         id: view
-        anchors.fill: parent
+        anchors.fill:parent
 
         PullDownMenu {
             id: pullDownMenu
@@ -69,7 +104,7 @@ Page {
 
         header: Item {
             width: view.width
-            height: dateHeader.height + (isPortrait ? datePicker.height : Theme.paddingLarge)
+            height: testColumn.height + (isPortrait ? datePicker.height : Theme.paddingLarge)
         }
 
         model: AgendaModel { id: agendaModel }
@@ -81,7 +116,7 @@ Page {
 
             x: isPortrait ? 0 : datePicker.width + Theme.paddingSmall
             width: isPortrait ? view.width : view.width - datePicker.width
-
+            y:isPortrait?datePicker.y + datePicker.height + testColumn.height: view.headerItem.y + Theme.paddingLarge + testColumn.height
             Component.onCompleted: activeDay = agendaModel.startDate
 
             Connections {
@@ -112,7 +147,9 @@ Page {
         }
 
         Column {
+            id:testColumn
             width: isPortrait ? view.width : view.width - datePicker.width
+            height:dateHeader.height + patchInfo.height + Theme.paddingLarge
             parent: view.contentItem
             x: isPortrait ? 0 : datePicker.width
             y: isPortrait ? datePicker.y + datePicker.height : view.headerItem.y + Theme.paddingLarge
@@ -122,6 +159,7 @@ Page {
                 width: parent.width
                 height: Screen.sizeCategory > Screen.Medium ? Theme.itemSizeMedium : Theme.itemSizeExtraSmall
                 onClicked: {
+                  console.log(testColumn.height)
                     var p = pageStack.push("DayPage.qml", { "width": root.width, "date": datePicker.date })
                     if (p !== null) {
                         p.statusChanged.connect(function() {
@@ -154,8 +192,59 @@ Page {
                                                                                     : Theme.primaryColor)
                 }
             }
-
+            Item{
+              id:patchInfo
+              width:parent.width
+              height:lunarItem.height+festivalItem.height + Theme.paddingMedium * 2
+              Item {
+                id:lunarItem
+                  anchors{
+                    top:dateHeader.bottom
+                    topMargin:Theme.paddingLarge
+                  }
+                  width: parent.width
+                  height: lunarText.height + Theme.paddingLarge
+                  visible: root.initialLoadDone
+                  Label {
+                      id: lunarText
+                      x: Theme.horizontalPageMargin
+                      anchors.verticalCenter: parent.verticalCenter
+                      width: parent.width - Theme.paddingLarge *2
+                      text:showLunarInfo()
+                      wrapMode: Text.WordWrap
+                      textFormat: Text.RichText;
+                      horizontalAlignment: Text.AlignRight
+                      font.pixelSize: Theme.fontSizeLarge
+                      color: Theme.secondaryHighlightColor
+                  }
+              }
+              Item {
+                  id:festivalItem
+                  anchors{
+                    top:lunarItem.bottom
+                    topMargin:Theme.paddingLarge
+                  }
+                  width: parent.width
+                  height: festivalText.txt ?festivalText.height + Theme.paddingLarge:0
+                  visible: root.initialLoadDone && festivalText.text.length > 6
+                  Label {
+                      id: festivalText
+                      x: Theme.horizontalPageMargin
+                      anchors.verticalCenter: parent.verticalCenter
+                      width: parent.width - Theme.paddingLarge *2
+                      text:showFestival()
+                      wrapMode: Text.WordWrap
+                      textFormat: Text.RichText;
+                      horizontalAlignment: Text.AlignRight
+                      font.pixelSize: Theme.fontSizeMedium
+                      color: Theme.secondaryHighlightColor
+                  }
+              }
+            }
             Item {
+              anchors{
+                top:patchInfo.bottom
+              }
                 width: parent.width
                 height: placeholderText.height + 2*Theme.paddingLarge
                 visible: view.count === 0 && root.initialLoadDone
@@ -175,5 +264,5 @@ Page {
             }
         }
     }
-}
 
+}
