--- /usr/share/jolla-calendar/pages/DatePickerPanel.qml
+++ /usr/share/jolla-calendar/pages/DatePickerPanel.qml
@@ -3,7 +3,7 @@ import Sailfish.Silica 1.0
 import org.nemomobile.calendar 1.0
 import org.nemomobile.time 1.0
 import "Util.js" as Util
-
+import "LunarCalendar.js" as Lunar
 Column {
     id: datePickerContainer
 
@@ -103,7 +103,12 @@ Column {
 
                     Text {
                         id: label
-                        anchors.centerIn: parent
+                        anchors{
+                            top:parent.top
+                            topMargin:Theme.paddingSmall
+                            horizontalCenter: parent.horizontalCenter
+
+                        }
                         text: model.day
                         font.pixelSize: Theme.fontSizeMedium
                         font.bold: model.day === wallClock.time.getDate()
@@ -122,6 +127,38 @@ Column {
                             return Theme.secondaryColor
                         }
                     }
+                    Label{
+                        id:lunarday
+                        anchors{
+                            top:label.bottom
+			    topMargin:Theme.paddingSmall*0.3
+                            horizontalCenter: parent.horizontalCenter
+                        }
+                        text: {
+                              var lunar = Lunar.LunarCalendar.solarToLunar(model.year, model.month+1, model.day)
+                              var lunarDay = lunar.lunarDay
+                              if(lunarDay == 1||lunarDay == "1"){
+                                  return lunar.lunarMonthName
+                              }
+                              return lunar.lunarDayName
+                            }
+                        font.pixelSize: Theme.fontSizeSmall*0.8
+                        font.bold: model.day === wallClock.time.getDate()
+                                    && model.month === wallClock.time.getMonth()+1
+                                    && model.year === wallClock.time.getFullYear()
+                        color: {
+                            if (model.day === datePicker.day &&
+                                model.month === datePicker.month &&
+                                model.year === datePicker.year) {
+                                return Theme.highlightColor
+                            } else if (lunarday.font.bold) {
+                                return Theme.highlightColor
+                            } else if (model.month === model.primaryMonth) {
+                                return Theme.primaryColor
+                            }
+                            return Theme.secondaryColor
+                        }
+                    }
 
                     Rectangle {
                         anchors.top: label.baseline
